### ISSUE TITLE
Ability to place specs to separate folder, angular-cli after eject webpack config

### ISSUE THREAD

**Issue by gorshkov-leonid** [angular/angular-cli#5865]
### Bug Report or Feature Request (mark with an `x`)
```
- [x ] bug report
- [ ?] feature request
```

### Versions.
```
@angular/cli: 1.0.0 (e)
@angular/common: 4.0.0
@angular/compiler: 4.0.0
@angular/core: 4.0.0
@angular/forms: 4.0.0
@angular/http: 4.0.0
@angular/platform-browser: 4.0.0
@angular/platform-browser-dynamic: 4.0.0
@angular/router: 4.0.0
@angular/cli: 1.0.0
@angular/compiler-cli: 4.0.0
node: 6.10.1
os: Windows 7 x64
```

### Repro steps.
0. ng new xxx, npm install
1. eject angular cli with options(it does not matter):
`ng eject --environment=prod --target=production --vendor-chunk=false --output-hashing=false --sourcemap=true --extract-css --verbose --sourcemap`
2. create sibling folder 'test' near with 'src'  and move 'src/test.ts' to 'test' folder.
3. replace path './src/test.ts' to './test/test.ts'
4. run npm run test
ER: success
AR: cannot build
```
Error: ENOENT: no such file or directory, stat '##################\src\tsconfig.spec.json'
    at Error (native)
    at Object.fs.statSync (fs.js:1000:11)
    at AotPlugin._setupOptions (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\@ngtools\webpack\src\plugin.js:58:16)
    at new AotPlugin (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\@ngtools\webpack\src\plugin.js:26:14)
    at _createAotPlugin (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\@angular\cli\models\webpack-configs\typescript.js:56:12)
    at Object.exports.getNonAotTestConfig (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\@angular\cli\models\webpack-configs\typescript.js:106:19)
    at WebpackTestConfig.buildConfig (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\@angular\cli\models\webpack-test-config.js:17:31)
    at init (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\@angular\cli\plugins\karma.js:79:94)
    at Array.invoke (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\di\lib\injector.js:75:15)
    at Injector.get (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\di\lib\injector.js:48:43)
    at X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\karma\lib\server.js:143:20
    at Array.forEach (native)
    at Server._start (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\karma\lib\server.js:142:21)
    at Injector.invoke (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\di\lib\injector.js:75:15)
    at Server.start (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\karma\lib\server.js:103:18)
    at Object.exports.run (X:\_work\VNFDeploymentWizard3\VNF-Deployment-Wizard-UI\node_modules\karma\lib\cli.js:280:26)
```
Note: path to config was got from '.angular-cli.json', see '"testTsconfig": "tsconfig.spec.json"' related to 'src'. 
But I expect that after eject will not need to store '.angular-cli.json' file. If I remove it will be error during to run '@angular/cli' framework (see karma.conf.js).

**Comment by gorshkov-leonid** [comment]
I have WA: 
1. leave tsconfig.spec.json in src, and specify path with ../
2 change .angular-cli.json
In any way I have to store .angular-cli.json, I think '@angular/cli' framework can be moved to more public place '@ngtools/...' and relation with .angular-cli.json can be removed.

What do you think about it?

**Comment by xingguangsixian** [comment]
I have the same question, who have the right solution?please share with me your solution?

**Comment by captainkovalsky** [comment]
We also try to use this approach

**Comment by clydin** [comment]
eject functionality is no longer available in the latest version (6.0+).  For custom webpack configurations the following unofficial add-on is a suggested solution: https://github.com/manfredsteyer/ngx-build-plus
